### PR TITLE
Update packaging

### PR DESCRIPTION
**Pull request by mpougin**
@kjappelbaum these are some changes I made to make the package installable in my Python 3.10 environment and play nice with `pymatgen==2023.7.20` (not sure what the supported range is, but it's above `v2022.0.0` due to the [changed import statements](https://pymatgen.org/compatibility.html#recent-breaking-changes)). However, I think we can just leave the pymatgen specification to other dependencies, see below.

I've mainly updated the package to use `pyproject.toml`, condensing a lot of files into one place. I've also updated the requirements since:

1. The `pymatgen` dependency was duplicated.
2. `matminer` already has `pandas`, `scikit_learn` and `pymatgen` as dependencies.

I've also adapted the tests so the ones that rely on the CSD Python API are skipped in case the package is not installed.

### CODE REVIEW

**Comment by mpougin** [comment]
Added a `tox` configuration, and locally the install/tests seem to run fine for Python 3.8-3.11:

```
 py38: OK (8.65=setup[1.58]+cmd[7.07] seconds)
  py39: OK (8.41=setup[1.43]+cmd[6.98] seconds)
  py310: OK (8.56=setup[1.37]+cmd[7.18] seconds)
  py311: OK (8.06=setup[1.37]+cmd[6.69] seconds)
  congratulations :) (33.72 seconds)
```

It failed to install for Python 3.7 since several dependencies down the line required Python>=3.8, so I've adapted our required Python version and removed the classifier.

**Comment by kjappelbaum** [comment]
Thank you! 🔥 💯 👍🏽

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+[build-system]
+requires = ['flit_core >=3.4,<4']
+build-backend = 'flit_core.buildapi'
+
+[project]
+name = "oximachine-featurizer"
+dynamic = ['version']
+description = "Mine MOF oxidation states and featurize metal sites."
+authors = [
+    { name = 'Kevin M. Jablonka' },
```
Comment (kjappelbaum) [inline pyproject.toml:10]: add yourself here, you thought everything!

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    "Programming Language :: Python :: 3.8",
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
+]
+requires-python = '>=3.8'
+dependencies = [
+    'ase~=3.0',
+    'numeral~=0.1',
+    'apricot-select~=0.6',
```
Comment (kjappelbaum) [inline pyproject.toml:31]: @kjappelbaum this can also me an optional dependency

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    'tqdm~=4.0',
+    'click~=8.1',
+    'matminer~=0.6',
+    'scikit-multilearn~=0.2',
```
Comment (kjappelbaum) [inline pyproject.toml:35]: i think this, too

File: test/parser/test_parser.py [hunk 4]
```diff
@@ -4,10 +4,16 @@
...
 
-from oximachine_featurizer.parse import GetOxStatesCSD
+try:
+    from oximachine_featurizer.parse import GetOxStatesCSD
+    skip_tests = False
+except ModuleNotFoundError:
+    skip_tests = True
 
 
+@pytest.mark.skipif(skip_tests, reason="This test can only run if the CSD Python API is installed.")
```
Comment (kjappelbaum) [inline test/parser/test_parser.py:16]: nice, thanks!

**Comment by mpougin** [comment]
thanks for the fast review @kjappelbaum

**Pull request merged**